cpp/src/arrow/extension/tensor_extension_array_test.cc

```diff
@@ -165,47 +170,51 @@ TEST_F(TestExtensionType, CreateFromArray) {
   ASSERT_EQ(ext_arr->null_count(), 0);
 }
 
+template <typename T>
 void CheckSerializationRoundtrip(const std::shared_ptr<DataType>& ext_type) {
-  auto fst_type = internal::checked_pointer_cast<FixedShapeTensorType>(ext_type);
-  auto serialized = fst_type->Serialize();
+  auto type = internal::checked_pointer_cast<T>(ext_type);
```

Comment:
Is this required? You're only calling methods on `ExtensionType`, so this cast should be sufficient:
```suggestion
  auto type = internal::checked_pointer_cast<ExtensionType>(ext_type);
```